Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,205 @@
+        Tuple2(Vector9(0,0,0,0,0,0,0,-1,0), Vector9(0,0,0,0,0,0,Rij.s10,Rij.s11,Rij.s12)),
+        Tuple2(Vector9(0,0,0,0,0,0,0,0,-1), Vector9(0,0,0,0,0,0,Rij.s20,Rij.s21,Rij.s22))
+      ])
+      
+      let b = Vector9(0, 0, 0, 0, 0, 0, 0, 0, 0)
+      
+      linearGraph.store(JacobianFactor9x9_2(jacobian: M9, error: b, edges: Tuple2(key1, key2)))
+    }
+    
+    let I_9x9: Jacobian9x9_1 = Array9([
+      Tuple1(Vector9(1,0,0,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,1,0,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,1,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,1,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,1,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,1,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,1,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,0,1,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,0,0,1))
+    ])
+    
+    // prior on the anchor orientation
+    linearGraph.store(JacobianFactor9x9_1(jacobian: I_9x9,
+                                          error: Vector9(1.0, 0.0, 0.0, /*  */ 0.0, 1.0, 0.0, /*  */ 0.0, 0.0, 1.0),
+                                          edges: Tuple1(associations[anchorId.perTypeID]!)))
+    
+    return linearGraph
+  }
+  
+  /// This function finds the closest Rot3 to the unconstrained 3x3 matrix with SVD.
+  /// TODO(fan): replace this with a 3x3 specialized SVD instead of this generic SVD (slow)
+  public func normalizeRelaxedRotations(
+    _ relaxedRot3: VariableAssignments,
+    associations: Dictionary<Int, TypedID<Vector9, Int>>,
+    ids: Array<TypedID<Pose3, Int>>) -> VariableAssignments {
+    var validRot3 = VariableAssignments()
+    
+    for v in ids {
+      let M_v: Vector9 = relaxedRot3[associations[v.perTypeID]!]
+      
+      let M = M_v.tensor.reshaped(to: [3, 3])
+      
+      let (_, U, V) = M.transposed().svd(computeUV: true, fullMatrices: true)
+      let UVT: Tensor<Double> = matmul(U!, V!.transposed())
+      
+      let det = (UVT[0, 0].scalar! * (UVT[1, 1].scalar! * UVT[2, 2].scalar! - UVT[2, 1].scalar! * UVT[1, 2].scalar!)
+                 - UVT[1, 0].scalar! * (UVT[0, 1].scalar! * UVT[2, 2].scalar! - UVT[2, 1].scalar! * UVT[0, 2].scalar!)
+                 + UVT[2, 0].scalar! * (UVT[0, 1].scalar! * UVT[1, 2].scalar! - UVT[1, 1].scalar! * UVT[0, 2].scalar!))
+      
+      let R = matmul(matmul(U!, Tensor<Double>(shape: [3], scalars: [1, 1, det]).diagonal()), V!.transposed()).transposed()
+      
+      let initRot = Rot3(coordinate:
+                          Matrix3Coordinate(Matrix3(R[0, 0].scalar!, R[0, 1].scalar!, R[0, 2].scalar!,
+                                                    R[1, 0].scalar!, R[1, 1].scalar!, R[1, 2].scalar!,
+                                                    R[2, 0].scalar!, R[2, 1].scalar!, R[2, 2].scalar!))
+      )
+      
+      // TODO(fan): relies on the assumption of continous and ordered allocation
+      let _ = validRot3.store(initRot)
+    }
+    return validRot3;
+  }
+  
+  /// This function computes the rotations
```

Comment:
Please expand this comment such as to make it useful

Reply:
done